### PR TITLE
[GH-13112] Add unit tests to "CommandListCmd" to use testify

### PR DESCRIPTION
**Pull request by 3mard**
Add unit test for commandListCmd 

fixes : https://github.com/mattermost/mattermost-server/issues/13112

### CODE REVIEW

File: commands/command_test.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
+package commands
+
+import (
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/mattermost/mmctl/printer"
+	"github.com/spf13/cobra"
+)
+
+func (s *MmctlUnitTestSuite) TestCommandListCmdF() {
+
```
Comment (mgdelacroix) [inline commands/command_test.go]: nit: unnecessary whitespace

File: commands/command_test.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+		s.client.EXPECT().GetAllTeams("", 0, 10000).Return(teams, &model.Response{Error: nil}).Times(1)
+		s.client.EXPECT().ListCommands(team1ID, true).Return(team1Commands, &model.Response{Error: nil}).Times(1)
+		s.client.EXPECT().ListCommands(team2Id, true).Return(team2Commands, &model.Response{Error: nil}).Times(1)
+		err := listCommandCmdF(s.client, cmd, []string{})
+		s.Len(printer.GetLines(), 2)
+		s.Equal(team1Commands[0], printer.GetLines()[0])
+		s.Equal(team2Commands[0], printer.GetLines()[1])
+		s.Nil(err)
+		s.Len(printer.GetErrorLines(), 0)
+
```
Comment (mgdelacroix) [inline commands/command_test.go]: nit: unnecessary whitespace

File: commands/command_test.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+		cmd := &cobra.Command{}
+		s.client.EXPECT().GetTeam(teamID, "").Return(team, &model.Response{Error: nil}).Times(1)
+		s.client.EXPECT().ListCommands(teamID, true).Return(teamCommand, &model.Response{Error: nil}).Times(1)
+		err := listCommandCmdF(s.client, cmd, []string{teamID})
+		s.Len(printer.GetLines(), 1)
+		s.Equal(teamCommand[0], printer.GetLines()[0])
+		s.Nil(err)
+		s.Len(printer.GetErrorLines(), 0)
+
```
Comment (mgdelacroix) [inline commands/command_test.go]: nit: unnecessary whitespace

File: commands/command_test.go [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+		cmd := &cobra.Command{}
+		team := &model.Team{Id: teamID}
+		s.client.EXPECT().GetTeam(teamID, "").Return(team, &model.Response{Error: nil}).Times(1)
+		s.client.EXPECT().ListCommands(teamID, true).Return(nil, &model.Response{Error: &model.AppError{}}).Times(1)
+		err := listCommandCmdF(s.client, cmd, []string{teamID})
+		s.Nil(err)
+		s.Len(printer.GetLines(), 0)
+		s.Len(printer.GetErrorLines(), 1)
+		s.Equal("Unable to list commands for '"+teamID+"'", printer.GetErrorLines()[0])
+
```
Comment (mgdelacroix) [inline commands/command_test.go]: nit: unnecessary whitespace

File: commands/command_test.go [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+		cmd := &cobra.Command{}
+		s.client.EXPECT().GetAllTeams("", 0, 10000).Return(teams, &model.Response{Error: nil}).Times(1)
+		s.client.EXPECT().ListCommands(team1ID, true).Return(team1Commands, &model.Response{Error: nil}).Times(1)
+		s.client.EXPECT().ListCommands(team2Id, true).Return(team2Commands, &model.Response{Error: nil}).Times(1)
+		err := listCommandCmdF(s.client, cmd, []string{})
+		s.Len(printer.GetLines(), 2)
+		s.Equal(team1Commands[0], printer.GetLines()[0])
+		s.Equal(team2Commands[0], printer.GetLines()[1])
+		s.Nil(err)
```
Comment (mgdelacroix) [inline commands/command_test.go]: nit: I'd push up this check to be the first one and modify it to be a required, as having an error here will invalidate the other checks and may cause an index out of range check

```suggestion
		s.Require().Nil(err)
```

File: commands/command_test.go [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+			},
+		}
+
+		cmd := &cobra.Command{}
+		s.client.EXPECT().GetTeam(teamID, "").Return(team, &model.Response{Error: nil}).Times(1)
+		s.client.EXPECT().ListCommands(teamID, true).Return(teamCommand, &model.Response{Error: nil}).Times(1)
+		err := listCommandCmdF(s.client, cmd, []string{teamID})
+		s.Len(printer.GetLines(), 1)
+		s.Equal(teamCommand[0], printer.GetLines()[0])
+		s.Nil(err)
```
Comment (mgdelacroix) [inline commands/command_test.go]: Same here, the `err` check should be the first one

```suggestion
		s.Require().Nil(err)
```

File: commands/command_test.go [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+		s.Equal(team2Commands[0], printer.GetLines()[1])
+		s.Nil(err)
+		s.Len(printer.GetErrorLines(), 0)
+
+	})
+
+	s.Run("List commands for a specific team", func() {
+		printer.Clean()
+		teamID := "team-id"
+
```
Comment (mgdelacroix) [inline commands/command_test.go]: nit: unnecessary whitespace

File: commands/command_test.go [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+		teamID := "non-existing-team"
+		printer.Clean()
+		cmd := &cobra.Command{}
+
+		// first try to get team by id
+		s.client.EXPECT().GetTeam(teamID, "").Return(nil, &model.Response{Error: nil}).Times(1)
+		// second try to search the team by name
+		s.client.EXPECT().GetTeamByName(teamID, "").Return(nil, &model.Response{Error: nil}).Times(1)
+		err := listCommandCmdF(s.client, cmd, []string{teamID})
+		s.Nil(err)
```
Comment (mgdelacroix) [inline commands/command_test.go]: Same case for the `Require()` here

File: commands/command_test.go [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+
+	s.Run("Failling to list commands for an existing team", func() {
+		teamID := "team-id"
+		printer.Clean()
+		cmd := &cobra.Command{}
+		team := &model.Team{Id: teamID}
+		s.client.EXPECT().GetTeam(teamID, "").Return(team, &model.Response{Error: nil}).Times(1)
+		s.client.EXPECT().ListCommands(teamID, true).Return(nil, &model.Response{Error: &model.AppError{}}).Times(1)
+		err := listCommandCmdF(s.client, cmd, []string{teamID})
+		s.Nil(err)
```
Comment (mgdelacroix) [inline commands/command_test.go]: Same case for the `Require()` here

**Comment by 3mard** [comment]
@mgdelacroix Thanks for the quick feedback. :)
